PR: Add group display name to Intune Assignment (Batch 1)

William-Francillette opened the pull request:
#### Pull Request (PR) description
<!--
    Replace this comment block with a description of your PR.
-->

#### This Pull Request (PR) fixes the following issues
<!--
    If this PR does not fix an open issue, replace this comment block with None.
    If this PR resolves one or more open issues, replace this comment block with
    a list the issues using a GitHub closing keyword, e.g.:
    - Fixes #123
    - Fixes #124
-->
* IntuneAccountProtectionLocalAdministratorPasswordSolutionPolicy
  * Added support for assignment GroupDisplayName
* IntuneAccountProtectionLocalUserGroupMembershipPolicy
  * Added support for assignment GroupDisplayName
* IntuneAccountProtectionPolicy
  * Added support for assignment GroupDisplayName
* IntuneAntivirusPolicyWindows10SettingCatalog
  * Added support for assignment GroupDisplayName
* IntuneAppConfigurationPolicy
  * Added support for assignment GroupDisplayName
* IntuneApplicationControlPolicyWindows10
  * Added support for assignment GroupDisplayName
* IntuneASRRulesPolicyWindows10
  * Added support for assignment GroupDisplayName
* IntuneDeviceCompliancePolicyAndroid
  * Added support for assignment GroupDisplayName
* IntuneDeviceCompliancePolicyAndroidDeviceOwner
  * Added support for assignment GroupDisplayName
* IntuneDeviceCompliancePolicyAndroidWorkProfile
  * Added support for assignment GroupDisplayName
* IntuneDeviceCompliancePolicyiOs
  * Added support for assignment GroupDisplayName
* IntuneDeviceCompliancePolicyMacOS
  * Added support for assignment GroupDisplayName
* IntuneDeviceCompliancePolicyWindows10
  * Added support for assignment GroupDisplayName

William-Francillette commented:
@NikCharlebois @ykuijs - not sure why but the test fails due to function not found but added to M365DSCUtil module so until it gets merged it will fail

ricmestre commented on Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAppConfigurationPolicy/MSFT_IntuneAppConfigurationPolicy.psm1:
```diff
@@ -634,6 +623,7 @@ function ConvertTo-M365DSCIntuneAppConfigurationPolicyCustomSettings
         $result += $currentSetting
     }
     return $result
+    Remove-M365DSCAuthenticationParameter
```
This one was most likely a copy pasto, please remove it :)

William-Francillette replied:
Good catch - not sure where that came from ...

ykuijs commented on Modules/Microsoft365DSC/DSCResources/MSFT_IntuneAccountProtectionPolicy/MSFT_IntuneAccountProtectionPolicy.psm1:
```diff
@@ -144,7 +144,7 @@ function Get-TargetResource
         #Retrieve policy general settings
 
-        $policy = Get-MgBetaDeviceManagementIntent -DeviceManagementIntentId $Identity -ErrorAction SilentlyContinue
+        $policy = Get-MgBetaDeviceManagementIntent -DeviceManagementIntentId $Identity -ExpandProperty settings,assignments -ErrorAction SilentlyContinue
```
Just as a check:
A lot of other resources have the Group.Read.All permissions added. This resource and the IntuneAntivirusPolicyWindows10SettingCatalog don't have that. Is that correct?

ykuijs commented on Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1:
```diff
@@ -4403,6 +4403,76 @@ function Remove-M365DSCAuthenticationParameter
+        }
+    }
+
+    return $validAuthenticationParameter
```
Wondering what this function is used for? Not sure if I understand the return value correctly.

William-Francillette replied:
updated variable name

ykuijs reviewed (changes requested):
Two comments/questions.

Also, not sure if I like the change of returning the nullresult to throwing an exception. Continuing the discussion in the other issue first.

ykuijs commented:
Now understand what you mean with the comment about the alternative solution being implemented in the issue. Didn't find it right away in this PR, but overlooked the comment that you only implemented it on one resource 😉 

Will add a comment to the issue.

ykuijs commented:
Quick comment: Please also correct the changelog. Currently there are two Unreleased sections.

William-Francillette commented:
Added support for groupDisplayName for an additional 7 resources
Also added groupDisplayName for the All Users and All Devices assignments (M365DSCDRGUtil: ConvertFrom-IntunePolicyAssignment)
